model-optimizer/extensions/front/tf/ObjectDetectionAPI.py

```diff
@@ -1185,14 +1198,162 @@ def generate_sub_graph(self, graph: Graph, match: SubgraphMatch):
                                                 dict(name='proposals'))
-
         # models with use_matmul_crop_and_resize = True should not swap order of elements (YX to XY) after the Proposal
         swap_proposals = not match.custom_replacement_desc.custom_attributes.get('do_not_swap_proposals', False) and \
                          not pipeline_config.get_param('use_matmul_crop_and_resize')
-
         if swap_proposals:
             proposal_node = add_convolution_to_swap_xy_coordinates(graph, proposal_node, 5)
 
+        return {'proposal_node': ObjectDetectionAPIProposalReplacement.ie_to_tf_proposals(graph, proposal_node, match,
+                                                                                          max_proposals,
+                                                                                          swap_proposals)}
+
+    @staticmethod
+    def insert_detection_output_instead_of_proposal(graph: Graph, match: SubgraphMatch,
+                                                    pipeline_config: PipelineConfig):
+        """
+        The function inserts DetectionOutput operation instead of Proposal operation which may result in an increase of
+        the accuracy for some models. The function is enabled with the custom attribute "operation_to_insert" with
+        value "DetectionOutput" in the transformation configuration file section for the
+        "ObjectDetectionAPIProposalReplacement" transformation. However, this transformation should not be applied in
+        case when an input image should be scaled before feeding the IR because the DetectionOutput operation does not
+        have information about the original input image size, whilst the Proposal operation has an input which contains
+        this information.
+        :param graph: the graph to operate on
+        :param match: the object containing information about the matched sub-graph
+        :param pipeline_config: object containing information from the pipeline.config file of the model
+        :return: the dictionary with mapping information needed for other transformations
+        """
+        max_proposals = _value_or_raise(match, pipeline_config, 'first_stage_max_proposals')
+
+        # Convolution/matmul node that produces classes predictions
+        # Transpose result of the tensor with classes permissions so it will be in a correct layout for Softmax
+        predictions_node = backward_bfs_for_operation(match.single_input_node(1)[0], ['Add'])[0]
+
+        ########################################################################################
+        # prepare input with class probabilities
+        reshape_classes_node = create_op_node_with_second_input(graph, Reshape, int64_array([0, -1, 2]),
+                                                                dict(name='predictions/Reshape'))
+        # transpose from NCHW to NHWC will be inserted as input to the Reshape automatically. This is expected
```

Comment:
Does it mean that it does not work if `--disable_nhwc_to_nchw` is turned on? Do we track it?